ci/infra/testrunner/utils/utils.py

```diff
@@ -139,14 +119,36 @@ def rsync(self, ip_address, remote_dir_path, local_dir_path):
                f'{local_dir_path}')
         self.runshellcommand(cmd)
 
-    def runshellcommand_withoutput(self, cmd, ignore_errors=True):
-        p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    def runshellcommand(self, cmd, cwd=None, env=None, ignore_errors=False):
+        """Running shell command in {workspace} if cwd == None
+           Eg) cwd is "skuba", cmd will run shell in {workspace}/skuba/
+               cwd is None, cmd will run in {workspace}
+               cwd is abs path, cmd will run in cwd
+        Keyword arguments:
+        cmd -- command to run
+        cwd -- dir to run the cmd
+        env -- environment variables
+        ignore_errors -- don't rise exception if command fails
+        """
+        if not cwd:
+            cwd = self.conf.workspace
+
+        if not os.path.isabs(cwd):
+            cwd = os.path.join(self.conf.workspace, cwd)
+
+        if not os.path.exists(cwd):
+            raise Exception(Format.alert("Directory {} does not exists".format(cwd)))
+
+        logger.info("Executing command: {} > {}".format(cwd, cmd))
+        p = subprocess.Popen(cmd, env=env, cwd=cwd, shell=True,
+                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Making stdout and stderr PIPEs is the behavior caused by `capture_output=True`, but specifying capture_output might be a little more readable.  That would mean using subprocess.run instead of directly calling subprocess.Popen, which is "only" Python 3.5 and newer - but SLES 15.1 is on 3.6.5, so that should be safe.  The other arguments in use are the same, so it'd be a drop-in - and the docs suggest that using `.run` is preferable to `.Popen` when possible.  It's up to you, though; just something to consider. :)

https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module